lib/fog/ovirt/requests/compute/v4/create_vm.rb

```diff
@@ -3,24 +3,25 @@ module Compute
     class Ovirt
       class V4
         class Real
-          def check_for_option(opts, name)
-            opts[name.to_sym] || opts[(name + "_name").to_sym]
-          end
 
           # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
           def process_vm_opts(opts)
-            return unless check_for_option(opts, "template") && check_for_option(opts, "storagedomain")
+            disk_attachments = []
+            opts[:disks].each do |disk_from_attachment|
+              sparse = disk_from_attachment["sparse"].present? ? disk_from_attachment["sparse"] == "true" : true
 
-            template_id = opts[:template] || client.system_service.templates_service.search(:name => opts[:template_name]).first.id
-            template_disks = client.system_service.templates_service.template_service(template_id).get.disk_attachments
-            storagedomain_id = opts[:storagedomain] || storagedomains.select { |s| s.name == opts[:storagedomain_name] }.first.id
+              disk = {
+                  :id => disk_from_attachment["id"],
+                  :format => disk_from_attachment["format"]|| OvirtSDK4::DiskFormat::COW,
+                  :sparse => sparse,
+                  :storage_domains => [client.system_service.storage_domains_service.storage_domain_service(disk_from_attachment["storage_domain"]).get]
+              }
 
-            # Make sure the 'clone' option is set if any of the disks defined by
-            # the template is stored on a different storage domain than requested
-            opts[:clone] = true unless opts[:clone] == true || template_disks.empty? || template_disks.all? { |d| d.storage_domain == storagedomain_id }
+                disk_attachments << OvirtSDK4::DiskAttachment.new(:disk => disk)
+              end
+            opts[:disk_attachments] = disk_attachments
 
-            # Create disks map
-            opts[:disks] = template_disks.collect { |d| { :id => d.id, :storagedomain => storagedomain_id } }
+            opts.delete(:disks)
```

Comment:
The function will return the contents of `opts[:disks]`. I think it would be better to return the whole `opts` hash.